src/layouts/Header.tsx

```diff
@@ -124,10 +126,11 @@ class Header extends React.Component<IProps, null> {
   }
 
   public handleConnect = async (_event: any): Promise<void> => {
-    enableWalletProvider({
+    await enableWalletProvider({
       suppressNotifyOnSuccess: true,
       showNotification: this.props.showNotification,
     });
+    this.forceUpdate();
```

Comment:
Hmm, `forceUpdate` is usually an antipattern.  Why is it needed?  Usually means some dependencies aren't right.

Reply:
I added this because the UI wasn't updating when you logged in. Not sure why it's needed here and not in the main app.

I don't know why it would update after the call to enableWalletProvider, do you know why it would update without a forceupdate?

Reply:
Which part of the UI is not updating?

The header should update (Connect button, account dropdown, etc)

If the `currentAccountAddress` has changed, then any part of the UI that depends on that should update.  See `pollForAccountChanges` and:  https://github.com/daostack/alchemy/blob/41f397bd0d9c494024715c21bc457d0adc2a88cc/src/layouts/AppContainer.tsx#L133

forceUpdate should not be required.

Reply:
Yep, the case I'm talking about is when the current address is already know (from local storage), but a new connection has been made to a provider.

Reply:
OK, then it is just the header we are concerned with updating?  If so then the dependencies in the header component are broken and need to be fixed between there and arc.ts, not by calling `forceUpdate`.

Reply:
For some reason this isn't showing up as a change request.  Adding a comment to try to force it.